ptbcontrib/extended_keyboards/keyboards.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+#
+# A library containing community-based extension for the python-telegram-bot library
+# Copyright (C) 2020-2025
+# The ptbcontrib developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+
+"""This module contains extended keyboard classes with extra functionality for PTB keyboards."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+from telegram import InlineKeyboardButton, InlineKeyboardMarkup
+
+if TYPE_CHECKING:
+    pass
+
+
+class IExtendedInlineKeyboardMarkup(
+    ABC,
+):
```

Comment:
```suggestion
class IExtendedInlineKeyboardMarkup(ABC):
```

what does the leading "I" stand for? Can't be "inline" b/c that's already included later on :D

Also what's the purpose of providing this abstract base class? You provide exactly one implementation so you could just drop the base class. Note that you don't even make this class public as it's not declared in `__init__`

Reply:
"I" - Interface.
The interface is here just in case, some maintainers sometimes require them, depend on the ptbcontrlib policy :)

Reply:
Ah okay. This reminds me of C++ style. I personally don't see much benefit of it...